table/tables/tables.go

```diff
@@ -1729,10 +1729,24 @@ func TryGetHandleRestoredDataWrapper(t table.Table, row []types.Datum, rowMap ma
 		}
 	}
 
-	for _, idx := range t.Meta().Indices {
-		if idx.Primary {
-			tablecodec.TruncateIndexValues(t.Meta(), idx, rsData)
-			break
+	// Try to truncate index values.
+	// Says that primary key(a (8)),
+	// For index t(a), don't truncate the value.
+	// For index t(a(9)), truncate to a(9).
+	// For index t(a(7)), truncate to a(8).
+	pkIdx := FindPrimaryIndex(t.Meta())
```

Comment:
Is it possible `pkIdx` is nil?

Reply:
No
if !collate.NewCollationEnabled() || !t.Meta().IsCommonHandle || t.Meta().CommonHandleVersion == 0 {
		return nil
	}